From 3405bf6f7413dfafa5d25f0b8f92f93160f382b4 Mon Sep 17 00:00:00 2001
From: Ashwin H <Ashwin.H@amd.com>
Date: Wed, 5 Jul 2023 07:04:49 +0000
Subject: [PATCH 27/30] hwmon: (pmbus/tps53679) Add support for TI TPS53659

Add support for TI TPS53659 controller to the tps53679 pmbus driver

Signed-off-by: Ashwin H <Ashwin.H@amd.com>
---
 drivers/hwmon/pmbus/tps53679.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/hwmon/pmbus/tps53679.c b/drivers/hwmon/pmbus/tps53679.c
index db2bdf2a1..126388d86 100644
--- a/drivers/hwmon/pmbus/tps53679.c
+++ b/drivers/hwmon/pmbus/tps53679.c
@@ -16,7 +16,7 @@
 #include "pmbus.h"
 
 enum chips {
-	tps53647, tps53667, tps53679, tps53681, tps53688
+	tps53647, tps53659, tps53667, tps53679, tps53681, tps53688
 };
 
 #define TPS53647_PAGE_NUM		1
@@ -206,6 +206,11 @@ static int tps53679_probe(struct i2c_client *client)
 		info->pages = TPS53647_PAGE_NUM;
 		info->identify = tps53679_identify;
 		break;
+	case tps53659:
+		info->pages = TPS53679_PAGE_NUM;
+		info->identify = tps53679_identify;
+		info->func[0] |= PMBUS_HAVE_PIN;
+		break;
 	case tps53679:
 	case tps53688:
 		info->pages = TPS53679_PAGE_NUM;
@@ -227,6 +232,7 @@ static int tps53679_probe(struct i2c_client *client)
 static const struct i2c_device_id tps53679_id[] = {
 	{"tps53647", tps53647},
 	{"tps53667", tps53667},
+	{"tps53659", tps53659},
 	{"tps53679", tps53679},
 	{"tps53681", tps53681},
 	{"tps53688", tps53688},
@@ -238,6 +244,7 @@ MODULE_DEVICE_TABLE(i2c, tps53679_id);
 static const struct of_device_id __maybe_unused tps53679_of_match[] = {
 	{.compatible = "ti,tps53647", .data = (void *)tps53647},
 	{.compatible = "ti,tps53667", .data = (void *)tps53667},
+	{.compatible = "ti,tps53659", .data = (void *)tps53659},
 	{.compatible = "ti,tps53679", .data = (void *)tps53679},
 	{.compatible = "ti,tps53681", .data = (void *)tps53681},
 	{.compatible = "ti,tps53688", .data = (void *)tps53688},
-- 
2.17.1

